stat -f: no numeric filesystem type on QNX.

QNX's struct statvfs (toybox's statfs there, see 0001) has the filesystem
type as a name, f_basetype, which %T prints, but no numeric f_type:
%t prints 0.

Upstream-Status: QNX-specific

--- a/toys/other/stat.c
+++ b/toys/other/stat.c
@@ -142,7 +142,12 @@
   else if (type == 'd') out('u', statfs->f_ffree);
   else if (type == 'f') out('u', statfs->f_bfree);
   else if (type == 'l') out('d', pathconf(TT.file, _PC_NAME_MAX));
+#ifdef __QNX__
+  // QNX's statvfs has the type's name (%T) but no number for it.
+  else if (type == 't') out('x', 0);
+#else
   else if (type == 't') out('x', statfs->f_type);
+#endif
   else if (type == 'T') strout(fs_type_name(statfs));
   else if (type == 'i') {
     int *val = (int *) &statfs->f_fsid;
